File: api/src/models/post.ts
import { Table, Column, Model, DataType, PrimaryKey, AllowNull, ForeignKey, BelongsTo, HasMany } from 'sequelize-typescript';
import { PostType } from '../types/types';
import User from './user';
import Comment from './comment';

@Table({
  tableName: 'Posts',
  timestamps: true,
})

class Post extends Model<PostType> {
  @PrimaryKey
  @AllowNull(false)
  @Column({ type: DataType.UUID })
  id!: string

  @ForeignKey(() => User)
  @AllowNull(true)
  @Column({ type: DataType.UUID })
  user_id!: string;

  @AllowNull(false)
  @Column({ type: DataType.TEXT })
  subject!: string;

  @AllowNull(false)
  @Column({ type: DataType.TEXT })
  title!: string;

  @AllowNull(false)
  @Column({ type: DataType.TEXT })
  content!: string;

  @AllowNull(true)
  @Column({ type: DataType.TEXT })
  image_url!: string | null;

  @BelongsTo(() => User, { as: 'user' })
  user!: User

  @HasMany(() => Comment, { as: 'comments', onDelete: 'CASCADE' })
  comments!: Comment[]
}

export default Post;

File: mobile/lib/models/post.dart
import 'package:mobile/models/comment.dart';
import 'package:mobile/models/user.dart';

class Post {
  final String id;
  final String userid;
  final String subject;
  final String content;
  final String? imageUrl;
  final String createdAt;
  final String updatedAt;  
  final User user;
  final List<Comment> comments;


  Post({
    required this.id,
    required this.userid,
    required this.subject,
    required this.content,
    this.imageUrl,
    required this.createdAt,
    required this.updatedAt,
    required this.user,
    required this.comments
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      userid: json['user_id'],
      subject: json['subject'],
      content: json['content'],
      imageUrl: json['image_url'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
      user: User.fromJson(json['user']),
      comments: (json['comments'] as List).map((comment) => Comment.fromJson(comment)).toList()
    );
  }
}